pkg/volume/empty_dir/empty_dir.go

```diff
@@ -254,8 +255,36 @@ func (ed *emptyDir) setupTmpfs(dir string) error {
 		return nil
 	}
 
-	glog.V(3).Infof("pod %v: mounting tmpfs for volume %v", ed.pod.UID, ed.volName)
-	return ed.mounter.Mount("tmpfs", dir, "tmpfs", nil /* options */)
+	memoryLimits, err := ed.getMemoryLimits()
+	if err != nil {
+		return err
+	}
+	var options []string
+	if memoryLimits > 0 {
+		options = []string{fmt.Sprintf("size=%d", memoryLimits)}
+	}
+	glog.V(3).Infof("pod %v: mounting tmpfs for volume %v with options %v", ed.pod.UID, ed.volName, options)
+	return ed.mounter.Mount("tmpfs", dir, "tmpfs", options)
+}
+
+func (ed *emptyDir) getMemoryLimits() (int64, error) {
+	if string(qos.GetPodQOS(ed.pod)) == "Guaranteed" {
+		_, limits, err := v1.PodRequestsAndLimits(ed.pod)
+		if err != nil {
+			return 0, err
+		}
+		val := limits[v1.ResourceMemory]
+		return (&val).Value(), nil
+	}
+	// For all other QOS class use allocatable as limit.
+	allocatable, err := ed.plugin.host.GetNodeAllocatable()
```

Comment:
what happens when a container uses all the system memory in their container?  It will get OOM'ed but the tmpfs will still exist.  The memory will be reparented to the pod cgroup.  The system is OOM, but killing the process did no good.

Until we are able to trap OOM and kill a whole pod, including umounting that tmpfs, maybe we should limit this to request? and if not specified, something sane and small?